rules/autogen.bzl

```diff
@@ -60,25 +45,12 @@ def _hjson_c_header(ctx):
         ),
     ]
 
-autogen_hjson_c_header = rule(
-    implementation = _hjson_c_header,
+opentitan_ip_c_header = rule(
+    implementation = _opentitan_ip_c_header_impl,
+    doc = "Generate the C headers for an IP block as used in a top",
     attrs = {
-        "srcs": attr.label_list(allow_files = True),
-        "node": attr.string(
-            doc = "Register block node to generate",
-        ),
-        "alias": attr.label(
-            mandatory = False,
-            allow_single_file = True,
-            doc = "A path to an alias file",
-        ),
```

Comment:
The alias feature might be used out-of-tree, to replace some of those integrator-specific CSRs with the real definitions. Dropping support for it _might_ be a problem. See https://github.com/lowRISC/opentitan/pull/10982 for the motivations.

Probably the Rust generation should support it as well, but that's a separate problem, haha.

Reply:
Thanks for the explanation, I had no idea what aliases were (I could not find any example or documentation). I can add it back to the rule and add an `alias` attribute to the `opentitan_ip` macros. This way integrators can just edit the `defs.bzl` files to add the alias if them want to.

Reply:
This would be used in a separate header generation target out-of-tree, I think. In other words, in a separate bazel workspace, someone would write their own `opentitan_ip_c_header` target that uses our rule, and our rule's responsibility is simply to pass through the file to the command line option and into the sandbox.

I'll note that we have no uses of the alias attribute in our repo. One of our integrators might be able to help, but we probably ought to document this, haha.

Reply:
I have implemented the aliases. It's supported both in the `opentitan_ip_c_header` rule and in the `opentitan_ip` macro so that it's also possible to just add the alias attribute directly at the definition point.